src/pcap/client.go

```diff
@@ -244,18 +247,20 @@ func (c *Client) ReadCaptureResponse(stream API_CaptureClient, packetWriter *pca
 
 	done := make(chan struct{})
 	go func() {
+		defer close(done)
 		for {
 			res, err := stream.Recv()
 			if errors.Is(err, io.EOF) {
 				logger.Info("clean stop, done")
 				cancel(nil)
-				break
+				return
 			}
+
 			code := status.Code(err)
 			if code != codes.OK {
```

Comment:
Do we need to check for an `OK` status as well?

Reply:
@peanball Why do you think we need the status OK? We are checking here for on err from stream.Recv but if there are no errors, we only read the packet/msg